core/codegen/src/derive/from_param.rs

```diff
@@ -0,0 +1,42 @@
+use devise::Support;
+use proc_macro2::TokenStream;
+use quote::quote;
+use devise::*;
+use devise::ext::SpanDiagnosticExt;
+use crate::exports::*;
+
+pub fn derive_from_param(input: proc_macro::TokenStream) -> TokenStream {
+    DeriveGenerator::build_for(input, quote!(impl<'a> #_request::FromParam<'a>))
+        .support(Support::Enum)
+        .validator(ValidatorBuild::new()
+                   .fields_validate(|_, fields| {
+                       if !fields.is_empty() {
+                           return Err(fields.span().error("Only empty enums are accepted"));
```

Comment:
Maybe the wording here should be "only enums without variants are supported"?

Reply:
Probably `Only enums without data fields are supported`.

Reply:
I think the language I was looking for was getting away form "enums with foo" and instead going for "enum variants with foo," since that's the actual constant. We should also make sure the error is on the fields themselves.

Reply:
`We should also make sure the error is on the fields themselves.` Do you mean the error message?

Reply:
He means the associated span, and it looks like you already have it correct.